Kerberos.NET/Replay/TicketCacheBase.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using static Kerberos.NET.MemoryTicketCache;
+
+namespace Kerberos.NET
+{
+    public abstract class TicketCacheBase : ITicketCache, IDisposable
+    {
+        private readonly Task backgroundRunner;
+
+        public TicketCacheBase(ILoggerFactory logger)
+        {
+            this.Logger = logger.CreateLoggerSafe<TicketCacheBase>();
+
+            this.Cancellation = new CancellationTokenSource();
+
+            this.backgroundRunner = Task.Run(this.RunBackground, this.Cancellation.Token);
+        }
+
+        public bool RefreshTickets { get; set; }
+
+        public TimeSpan RefreshInterval { get; set; } = TimeSpan.FromSeconds(30);
+
+        internal Func<CacheEntry, Task> Refresh { get; set; }
+
+        protected CancellationTokenSource Cancellation { get; }
+
+        protected ILogger Logger { get; }
+
+        private async Task RunBackground()
+        {
+            while (true)
+            {
+                if (this.Cancellation.Token.IsCancellationRequested)
+                {
+                    break;
+                }
+
+                try
+                {
+                    await this.BackgroundCacheOperation().ConfigureAwait(true);
+                }
+                catch (Exception ex)
+                {
+                    this.Logger.LogWarning(ex, "Background cache operation failed");
+                }
+
+                await Task.Delay(this.RefreshInterval, this.Cancellation.Token).ConfigureAwait(true);
+            }
+        }
+
+        protected virtual Task BackgroundCacheOperation()
+        {
+            return Task.CompletedTask;
+        }
+
+        public void Dispose()
+        {
+            this.Cancellation.Cancel();
+            this.Cancellation.Dispose();
+
+            this.backgroundRunner.ContinueWith(t => t.Dispose(), TaskScheduler.Default);
```

Comment:
```suggestion
```
Not necessary, we can rely on the runtime to collect this one.
Cf. [Do I need to dispose of Tasks?](https://devblogs.microsoft.com/pfxteam/do-i-need-to-dispose-of-tasks/).

Reply:
There's some stylecop rules that are about to get applied across the entire project that make it easier to just call dispose.